src/bech32.cpp

```diff
@@ -370,11 +370,11 @@ std::string Encode(Encoding encoding, const std::string& hrp, const data& values
 }
 
 /** Decode a Bech32 or Bech32m string. */
-DecodeResult Decode(const std::string& str) {
+DecodeResult Decode(const std::string& str, CharLimit limit) {
     std::vector<int> errors;
     if (!CheckCharacters(str, errors)) return {};
     size_t pos = str.rfind('1');
-    if (str.size() > 90 || pos == str.npos || pos == 0 || pos + 7 > str.size()) {
+    if (str.size() > limit || pos == str.npos || pos == 0 || pos + 7 > str.size()) {
```

Comment:
Aren't these checks broken if limit is `NO_LIMIT` (0) ?

Reply:
:doh: yes! fixed and added unit tests for the `NO_LIMIT` case.